### PR TITLE
[GPU] Enable encryption of cache blob with CacheMode::OPTIMIZE_SIZE

### PR DESCRIPTION
**Pull request by tkrupa-intel**
### Details:
 - Enables encryption of cache blob with CacheMode::OPTIMIZE_SIZE in GPU Plugin.
 - Some additional test coverage already present in src/plugins/intel_gpu/tests/functional/shared_tests_instances/behavior/ov_plugin/caching_tests.cpp. Test coverage in this PR is distinct from those tests because it also checks correctness of the results.
 - #27742 has to be merged first - it guarantees small cache sizes in majority of cases which is important for encryption.

### Tickets:
 - CVS-158140

### CODE REVIEW

**Comment by e-ddykim** [comment]
Instead of encrypting or decrypting the entire blob all at once, can't we do it in multiple batches?

**Comment by tkrupa-intel** [comment]
@e-ddykim Before implementing I talked about this with @ilya-lavrenov and I decided to fulfill an expectation to give encryption algorithm developers a full freedom to analyze/encrypt the whole data. This is only enabled with CacheMode::OPTIMIZE_SIZE so with blobs sized a couple of MB (after the dependency PR is merged).

A second consideration - are we certain that breaking it up into arbitrarily sized blocks wouldn't compromise security of any encryption algorithm a user may provide?

**Review by e-ddykim — APPROVED** [review]

**Review by olpipi — APPROVED** [review]
LGTM

just question: Is this feature expected to work on all OSs and platforms?

**Comment by tkrupa-intel** [comment]
@olpipi Yes, that's the intention here.

**Review by sshlyapn — APPROVED** [review]

**Review by p-durandin — APPROVED** [review]

**Comment by tkrupa-intel** [comment]
build_jenkins

File: src/inference/include/openvino/runtime/properties.hpp [hunk 1]
```diff
@@ -801,6 +801,8 @@ struct EncryptionCallbacks {
  * when loading from the cache. This property is set in core.compile_model only.
  * - First value of the struct is encryption function.
  * - Second value of the struct is decryption function.
+ * @note GPU Plugin: encrypts whole blob, not only model structure. Only used when ov::CacheMode Property is set to
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/properties.hpp:804]:
```suggestion
 * @note GPU Plugin: encrypts whole blob, not only model structure. Only used when ov::cache_mode property is set to
```

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by tkrupa-intel** [comment]
build_jenkins

**Comment by tkrupa-intel** [comment]
build_jenkins

**Comment by tkrupa-intel** [comment]
build_jenkins

**Pull request merged**